### PR TITLE
1666: Rename CardBlueprint to Card

### PR DESCRIPTION
**Pull request by steffenkleinle**
### Short description

Follow up PR to #1667 to make the naming consistent.

### Proposed changes

- Rename everything to just `card`/`Card`.

### Side effects

None.

### Testing

Nothing breaks.

### Resolved issues

Fixes: #1666

### CODE REVIEW

**Review by ztefanie — APPROVED** [review]
Looks good, thank you :rocket: 

Tried to test, but i get this error, that seems unrelated to this PR, therefore did not test
```
/home/steffi/.nvm/versions/node/v16.13.2/bin/npm run start

> start
> set -ae; REACT_APP_APPLICATION_COMMIT=$(./application_commit.sh); ts-node scripts/start.ts
Starting the development server...
Failed to compile.

Module parse failed: Unexpected token (6:7)
File was processed with these loaders:
 * ../node_modules/@pmmmwh/react-refresh-webpack-plugin/loader/index.js
 * ../node_modules/source-map-loader/dist/cjs.js
You may need an additional loader to handle the result of these loaders.
| export { default as buildConfigBayern } from "./bayern"
| export { default as buildConfigKoblenz } from "./koblenz"
> export type { DeeplLinkingConfig, BuildConfigType } from "./types"
| export * from "./constants"
|
ERROR in ../frontend/build-configs/index.ts 6:7
Module parse failed: Unexpected token (6:7)
File was processed with these loaders:
 * ../node_modules/@pmmmwh/react-refresh-webpack-plugin/loader/index.js
 * ../node_modules/source-map-loader/dist/cjs.js
You may need an additional loader to handle the result of these loaders.
| export { default as buildConfigBayern } from "./bayern"
| export { default as buildConfigKoblenz } from "./koblenz"
> export type { DeeplLinkingConfig, BuildConfigType } from "./types"
| export * from "./constants"
| 

webpack compiled with 1 error
```

**Comment by steffenkleinle** [comment]
> Looks good, thank you 🚀
> 
> Tried to test, but i get this error, that seems unrelated to this PR, therefore did not test
> 
> ```
> /home/steffi/.nvm/versions/node/v16.13.2/bin/npm run start
> 
> > start
> > set -ae; REACT_APP_APPLICATION_COMMIT=$(./application_commit.sh); ts-node scripts/start.ts
> Starting the development server...
> Failed to compile.
> 
> Module parse failed: Unexpected token (6:7)
> File was processed with these loaders:
>  * ../node_modules/@pmmmwh/react-refresh-webpack-plugin/loader/index.js
>  * ../node_modules/source-map-loader/dist/cjs.js
> You may need an additional loader to handle the result of these loaders.
> | export { default as buildConfigBayern } from "./bayern"
> | export { default as buildConfigKoblenz } from "./koblenz"
> > export type { DeeplLinkingConfig, BuildConfigType } from "./types"
> | export * from "./constants"
> |
> ERROR in ../frontend/build-configs/index.ts 6:7
> Module parse failed: Unexpected token (6:7)
> File was processed with these loaders:
>  * ../node_modules/@pmmmwh/react-refresh-webpack-plugin/loader/index.js
>  * ../node_modules/source-map-loader/dist/cjs.js
> You may need an additional loader to handle the result of these loaders.
> | export { default as buildConfigBayern } from "./bayern"
> | export { default as buildConfigKoblenz } from "./koblenz"
> > export type { DeeplLinkingConfig, BuildConfigType } from "./types"
> | export * from "./constants"
> | 
> 
> webpack compiled with 1 error
> ```

Hmmm, interersting, did you try to run `npm i` and/or `npm run ts:check`? At least it looks like its not finding something related to build configs that should be there. For me it works without issues.

**Comment by ztefanie** [comment]
Yes i even deleted node_modules folder and reinstalled.

**Comment by michael-markl** [comment]
@ztefanie No idea if it's related, but your Node version seems outdated (v16). We use v20 in circle ci I think

File: administration/src/bp-modules/cards/ImportCardsInput.tsx [hunk 1]
```diff
@@ -87,12 +82,12 @@ const ImportCardsInput = ({
       }
 
       const csvHeader = lines.shift() ?? []
-      const blueprints = lines.map(line => lineToBlueprint(line, csvHeader))
+      const blueprints = lines.map(line => lineToCard(line, csvHeader))
```
Comment (f1sh1918) [inline administration/src/bp-modules/cards/ImportCardsInput.tsx:85]:
```suggestion
      const cards = lines.map(line => lineToCard(line, csvHeader))
```

File: administration/src/bp-modules/cards/ImportCardsInput.tsx [hunk 2]
```diff
@@ -87,12 +82,12 @@ const ImportCardsInput = ({
...
 
-      setCardBlueprints(blueprints)
+      setCards(blueprints)
```
Comment (f1sh1918) [inline administration/src/bp-modules/cards/ImportCardsInput.tsx:87]:
```suggestion
      setCards(cards)
```

**Review by f1sh1918 — COMMENTED** [review]
i think you missed the `AddCardsForm.tsx`

Found lots of blueprints there @steffenkleinle

File: administration/src/bp-modules/cards/AddCardsForm.tsx [hunk 3]
```diff
@@ -84,13 +76,13 @@ const AddCardsForm = ({
...
-    setCardBlueprints([...cardBlueprints, cardBlueprint])
+    const card = initializeCard(projectConfig.card, region)
+    setCards([...cards, card])
     scrollToBottom()
-  }, [cardBlueprints, projectConfig.card, region, setCardBlueprints])
+  }, [cards, projectConfig.card, region, setCards])
 
-  const removeCardBlueprint = (oldBlueprint: CardBlueprint) => {
-    setCardBlueprints(cardBlueprints.filter(blueprint => blueprint !== oldBlueprint))
+  const removeCard = (oldBlueprint: Card) => {
```
Comment (f1sh1918) [inline administration/src/bp-modules/cards/AddCardsForm.tsx:84]:
```suggestion
  const removeCard = (oldCard: Card) => {
```

File: administration/src/bp-modules/cards/AddCardsForm.tsx [hunk 4]
```diff
@@ -84,13 +76,13 @@ const AddCardsForm = ({
...
+    setCards(cards.filter(blueprint => blueprint !== oldBlueprint))
```
Comment (f1sh1918) [inline administration/src/bp-modules/cards/AddCardsForm.tsx:85]:
```suggestion
    setCards(cards.filter(card => card !== oldCard))
```

File: administration/src/bp-modules/cards/AddCardsForm.tsx [hunk 5]
```diff
@@ -99,12 +91,12 @@ const AddCardsForm = ({
...
           scrollToBottom()
         }}>
-        {cardBlueprints.map((blueprint, index) => (
+        {cards.map((blueprint, index) => (
           <FormColumn key={blueprint.id}>
             <AddCardForm
-              cardBlueprint={blueprint}
-              onRemove={() => removeCardBlueprint(blueprint)}
-              updateCard={updatedCard => updateCardBlueprint(updatedCard, index)}
+              card={blueprint}
```
Comment (f1sh1918) [inline administration/src/bp-modules/cards/AddCardsForm.tsx:97]:
```suggestion
              card={card}
```

File: administration/src/bp-modules/cards/AddCardsForm.tsx [hunk 6]
```diff
@@ -99,12 +91,12 @@ const AddCardsForm = ({
...
+              onRemove={() => removeCard(blueprint)}
```
Comment (f1sh1918) [inline administration/src/bp-modules/cards/AddCardsForm.tsx:98]:
```suggestion
              onRemove={() => removeCard(card)}
```

File: administration/src/bp-modules/cards/AddCardsForm.tsx [hunk 7]
```diff
@@ -99,12 +91,12 @@ const AddCardsForm = ({
         onFinishAll={() => {
           scrollToBottom()
         }}>
-        {cardBlueprints.map((blueprint, index) => (
+        {cards.map((blueprint, index) => (
           <FormColumn key={blueprint.id}>
```
Comment (f1sh1918) [inline administration/src/bp-modules/cards/AddCardsForm.tsx:95]:
```suggestion
          <FormColumn key={card.id}>
```

File: administration/src/bp-modules/cards/AddCardsForm.tsx [hunk 8]
```diff
@@ -37,27 +37,27 @@ const FormColumn = styled.div`
 
 type CreateCardsFormProps = {
   region: Region
-  cardBlueprints: CardBlueprint[]
-  setCardBlueprints: (blueprints: CardBlueprint[]) => void
-  updateCardBlueprint: (updatedCard: Partial<CardBlueprint>, index: number) => void
+  cards: Card[]
+  setCards: (blueprints: Card[]) => void
```
Comment (f1sh1918) [inline administration/src/bp-modules/cards/AddCardsForm.tsx:41]:
```suggestion
  setCards: (cards: Card[]) => void
```

**Review by f1sh1918 — APPROVED** [review]
I added a proposal pr for missing changes.
You can ignore the comments here.

Smoke tested the proposal so should be fine

**Pull request merged**